carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -197,104 +296,330 @@ std::shared_ptr<Geofence> WMBroadcaster::geofenceFromMsg(const cav_msgs::Traffic
+  for (auto regem : parallel_llt_front.regulatoryElements()) //copy existing regem into the new llts
+  {
+    front_llt_diag.addRegulatoryElement(regem);
   }
 
+  // create trapezoids on back side TODO: more sophisticated diagonal llt?
+  lanelet::LineString3d back_left_ls(lanelet::utils::getId(), { middle_opposite_lanelets->front().rightBound3d().front(), parallel_llt_back.leftBound3d().front()});  
+  lanelet::LineString3d back_right_ls(lanelet::utils::getId(), { middle_opposite_lanelets->front().leftBound3d().front(), parallel_llt_back.rightBound3d().front()});
+  lanelet::Lanelet back_llt_diag (lanelet::utils::getId(), back_left_ls, back_right_ls);
+  for (auto regem : parallel_llt_back.regulatoryElements()) //copy existing regem into the new llts
+  {
+    back_llt_diag.addRegulatoryElement(regem);
+  }
 
-  // If the days of the week are specified then convert them to the boost format. 
-  GeofenceSchedule::DayOfTheWeekSet week_day_set = { 0, 1, 2, 3, 4, 5, 6 }; // Default to all days  0==Sun to 6==Sat
-  if (msg_schedule.dow_exists) {
-   // sun (6), mon (5), tue (4), wed (3), thu (2), fri (1), sat (0)
-   week_day_set.clear();
-   for (size_t i = 0; i < msg_schedule.dow.dow.size(); i++) {
-     if (msg_schedule.dow.dow[i] == 1) {
-      switch(i) {
-          case 6: // sun
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Sunday);
-            break;
-          case 5: // mon
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Monday);
-            break;
-          case 4: // tue
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Tuesday);
-            break;
-          case 3: // wed
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Wednesday);
-            break;
-          case 2: // thur
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Thursday);
-            break;
-          case 1: // fri
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Friday);
-            break;
-          case 0: // sat
-            week_day_set.emplace(boost::gregorian::greg_weekday::weekday_enum::Saturday);
-            break;
-          default:
-            throw std::invalid_argument("Unrecognized weekday value: " + std::to_string(i));
-      }
-     }
-   }
+  // create new middle lanelets to match the parallel direction
+  std::vector<lanelet::Lanelet> middle_llts;
+  for (auto llt : *middle_opposite_lanelets.get())
+  {
+    lanelet::Lanelet middle_llt (lanelet::utils::getId(), llt.rightBound3d().invert(), llt.leftBound3d().invert());
+    for (auto regem : llt.regulatoryElements())
+    {
+      middle_llt.addRegulatoryElement(regem); //copy existing regem into the new llts
+    }
+    middle_llts.push_back(middle_llt);
   }
 
-  if (msg_schedule.between_exists) {
+  // add traffic light and stoprule to the lanelet of parallel direction
```

Comment:
These big code blocks below could use some white space to make them more readable. They are pretty hard to process as is

Reply:
added section comment and extra white lines

Reply:
I think you could add more white space between individual lines especially in the first two really dense blocks